src/mm2/mm2.client.hpp

```diff
@@ -197,28 +219,66 @@ namespace antara::mmbot
 
         void from_json(const nlohmann::json &j, buy_request &cfg);
 
-        struct buy_result
+        struct buy_result : trade_result
+        {
+            antara::side side = antara::side::buy;
+            // std::string action;
```

Comment:
If it's working, we can remove the commentary section

Reply:
You're happy with my approach?

Reply:
Yeah it's fine, There is 2 way's to do it, (inheritance or composition) you choose inheritance, is fine.